validator/client/src/main/java/tech/pegasys/teku/validator/client/ActiveKeyManager.java

```diff
@@ -160,6 +165,33 @@ public DeleteRemoteKeysResponse deleteExternalValidators(List<BLSPublicKey> vali
     return new DeleteRemoteKeysResponse(deletionResults);
   }
 
+  private void deleteExternalDoppelgangers(final Set<BLSPublicKey> publicKeys) {
```

Comment:
I'm not a fan of duplicating the deletion logic here. Given it's a rare situation and the implementation is fairly complex and important to get exactly right, we'd be better off reusing the existing code even if it performs additional computation.

Can probably call the existing `deleteExternalValidators` method and then checking if the result is successful or not and reporting from there.

Reply:
I refactored the existing deletion logic to pass on the public keys (The same for the import results) and reused it for both internal and external validators